### ISSUE TITLE
Add support for pdb

### ISSUE THREAD

**Issue by sanderr** [inmanta/inmanta-core#5634]
Compile errors can be difficult to debug. This ticket proposes a rough draft to hook into pdb when a compile error occurs. This would allow the model developer to more easily inspect the state of the model at the time where the compile error is raised.

# User interface
Add a `--pdb` option that opens a `pdb` session when a (certain class of?) compiler error(s) occurs. The user can access the exception object, which contains references to relevant model objects. Through those the user can inspect values and traverse relations.

# Implementation
- Start a `pdb` session when a compiler error occurs and the `--pdb` option is set. This is the trivial part.
- Wrap objects in `DynamicProxy` instances where appropriate, in order to present a plugin-like experience to the user.
- Nice-to-have: make sure all attributes and relations can be accessed, even if they have not been frozen yet. e.g. by freezing them before starting `pdb`, or by enabling a special mode that doesn't raise not set exceptions?
